src/elm/Update.elm

```diff
@@ -90,3 +101,21 @@ update msg model =
 
         TickEvents ->
             { model | events = Expirable.tickAll model.events } ! []
+
+        GetItem ->
+            model ! [ loadInventoryFromLocalStorage model ]
+
+        SetItem ->
+            model
+                ! [ setItem model.storage
+                        "inventory"
+                        (encodeInventory model.inventory)
+                  ]
+
+        UpdatePort value ->
+            { model | inventory = decodeInventory value } ! []
+
+
+loadInventoryFromLocalStorage : Model -> Cmd Msg
+loadInventoryFromLocalStorage { storage } =
+    getItem storage "inventory"
```

Comment:
Thoughts on extracting `"inventory"` to a separate function to share both here and above, in the pattern-match for `SetItem`?

Reply:
@joshuaclayton I'm not sure I follow.
what do you think about this?
```elm
        GetItem ->
            model ! [ loadInventoryFromLocalStorage model ]

        SetItem ->
            model ! [ saveInventoryToLocalStorage model ]

        UpdatePort value ->
            { model | inventory = decodeInventory value } ! []


loadInventoryFromLocalStorage : Model -> Cmd Msg
loadInventoryFromLocalStorage { storage } =
    getItem storage "inventory"


saveInventoryToLocalStorage : Model -> Cmd Msg
saveInventoryToLocalStorage { storage, inventory } =
    setItem storage "inventory" (encodeInventory inventory)
```